vendor: bump go.etcd.io/etcd

tbg opened the pull request:
This picks up Joint Consensus, see:

https://github.com/etcd-io/etcd/pull/10914

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/39046)
<!-- Reviewable:end -->

tbg committed:
vendor: bump go.etcd.io/etcd

This picks up Joint Consensus, see:

https://github.com/etcd-io/etcd/pull/10914

Release note: None

tbg commented:
bors r=nvanbenschoten
